dace/codegen/targets/intel_fpga.py

```diff
@@ -538,40 +536,39 @@ def generate_module(self, sdfg, state, name, subgraph, parameters,
                     "const int " + p for p in scope.params
                 ]  # PE id will be a macro defined constant
                 kernel_args_call += [p for p in scope.params]
-                unrolled_loops += 1
+                unrolled_loop = scope.map
 
         # Ensure no duplicate parameters are used
         kernel_args_opencl = dtypes.deduplicate(kernel_args_opencl)
         kernel_args_call = dtypes.deduplicate(kernel_args_call)
 
         # Add kernel call host function
-        if unrolled_loops == 0:
+        if unrolled_loop is None:
             host_body_stream.write(
                 "kernels.emplace_back(program.MakeKernel(\"{}\"{}));".format(
                     module_function_name, ", ".join([""] + kernel_args_call)
                     if len(kernel_args_call) > 0 else ""), sdfg, state_id)
         else:
             # We will generate a separate kernel for each PE. Adds host call
-            for ul in self._unrolled_pes:
```

Comment:
Can't we have nested maps that generates PE?

Reply:
Maybe, but the rest of the code you wrote assumes maximum one map :P